src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -109,7 +116,8 @@ public static FormValidation checkScanCredentials(SCMSourceOwner context, String
             if (!found) {
                 return FormValidation.error("Credentials not found");
             }
-            if (!(context.hasPermission(Item.CONFIGURE)
+            if (context != null && !(
```

Comment:
The main fix. Note that we already did a null/permission check at the start of this method.